In [1]:
import pandas as pd

In [57]:
#pd.read_excel("filename.xlsx")

In [ ]:
#df.to_excel("fghfg.xlsx")

In [2]:
df=pd.read_csv("https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv")
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

In [6]:
df.iloc[[1,4],[3,4]]

Name     Sex
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female
4                           Allen, Mr. William Henry    male

In [9]:
df.iloc[1:5,0:8]

PassengerId  Survived  Pclass  \
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch  
1      0  
2      0  
3      0  
4      0

In [72]:
df.iloc[0:5,3:5]

Name     Sex
0                            Braund, Mr. Owen Harris    male
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female
2                             Heikkinen, Miss. Laina  female
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female
4                           Allen, Mr. William Henry    male

In [73]:
df.iloc[[0,1,886],3:5]

Name     Sex
0                              Braund, Mr. Owen Harris    male
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female
886                              Montvila, Rev. Juozas    male

In [74]:
df.iloc[[0,1,886],[3,4,11]]

Name     Sex Embarked
0                              Braund, Mr. Owen Harris    male        S
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female        C
886                              Montvila, Rev. Juozas    male        S

In [12]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [14]:
df.tail()

PassengerId  Survived  Pclass                                      Name  \
886          887         0       2                     Montvila, Rev. Juozas   
887          888         1       1              Graham, Miss. Margaret Edith   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
889          890         1       1                     Behr, Mr. Karl Howell   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch      Ticket   Fare Cabin Embarked  
886    male  27.0      0      0      211536  13.00   NaN        S  
887  female  19.0      0      0      112053  30.00   B42        S  
888  female   NaN      1      2  W./C. 6607  23.45   NaN        S  
889    male  26.0      0      0      111369  30.00  C148        C  
890    male  32.0      0      0      370376   7.75   NaN        Q

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [6]:
df.dtypes

PassengerId      int64
Survived         int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
dtype: object

In [7]:
df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

In [21]:
df[df.dtypes[df.dtypes=="float64"].index]

Age     Fare
0    22.0   7.2500
1    38.0  71.2833
2    26.0   7.9250
3    35.0  53.1000
4    35.0   8.0500
..    ...      ...
886  27.0  13.0000
887  19.0  30.0000
888   NaN  23.4500
889  26.0  30.0000
890  32.0   7.7500

[891 rows x 2 columns]

In [20]:
df[df.dtypes[df.dtypes=="int64"].index]

PassengerId  Survived  Pclass  SibSp  Parch
0              1         0       3      1      0
1              2         1       1      1      0
2              3         1       3      0      0
3              4         1       1      1      0
4              5         0       3      0      0
..           ...       ...     ...    ...    ...
886          887         0       2      0      0
887          888         1       1      0      0
888          889         0       3      1      2
889          890         1       1      0      0
890          891         0       3      0      0

[891 rows x 5 columns]

In [19]:
df[df.dtypes[df.dtypes=="object"].index]

Name     Sex  \
0                              Braund, Mr. Owen Harris    male   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female   
2                               Heikkinen, Miss. Laina  female   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female   
4                             Allen, Mr. William Henry    male   
..                                                 ...     ...   
886                              Montvila, Rev. Juozas    male   
887                       Graham, Miss. Margaret Edith  female   
888           Johnston, Miss. Catherine Helen "Carrie"  female   
889                              Behr, Mr. Karl Howell    male   
890                                Dooley, Mr. Patrick    male   

               Ticket Cabin Embarked  
0           A/5 21171   NaN        S  
1            PC 17599   C85        C  
2    STON/O2. 3101282   NaN        S  
3              113803  C123        S  
4              373450   NaN        S  
..                ...   ...      ...  
886            211536   NaN        S  
887            112053   B42        S  
888        W./C. 6607   NaN        S  
889            111369  C148        C  
890            370376   NaN        Q  

[891 rows x 5 columns]

In [52]:
df[df.dtypes[df.dtypes=="object"].index]

Name     Sex  \
0                              Braund, Mr. Owen Harris    male   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female   
2                               Heikkinen, Miss. Laina  female   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female   
4                             Allen, Mr. William Henry    male   
..                                                 ...     ...   
886                              Montvila, Rev. Juozas    male   
887                       Graham, Miss. Margaret Edith  female   
888           Johnston, Miss. Catherine Helen "Carrie"  female   
889                              Behr, Mr. Karl Howell    male   
890                                Dooley, Mr. Patrick    male   

               Ticket Cabin Embarked  
0           A/5 21171   NaN        S  
1            PC 17599   C85        C  
2    STON/O2. 3101282   NaN        S  
3              113803  C123        S  
4              373450   NaN        S  
..                ...   ...      ...  
886            211536   NaN        S  
887            112053   B42        S  
888        W./C. 6607   NaN        S  
889            111369  C148        C  
890            370376   NaN        Q  

[891 rows x 5 columns]

In [54]:
df[df.dtypes[df.dtypes=="int64"].index]

PassengerId  Survived  Pclass  SibSp  Parch
0              1         0       3      1      0
1              2         1       1      1      0
2              3         1       3      0      0
3              4         1       1      1      0
4              5         0       3      0      0
..           ...       ...     ...    ...    ...
886          887         0       2      0      0
887          888         1       1      0      0
888          889         0       3      1      2
889          890         1       1      0      0
890          891         0       3      0      0

[891 rows x 5 columns]

In [53]:
df.dtypes

PassengerId      int64
Survived         int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
dtype: object

In [47]:
df[df.dtypes[df.dtypes=="object"].index]

Name     Sex  \
0                              Braund, Mr. Owen Harris    male   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female   
2                               Heikkinen, Miss. Laina  female   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female   
4                             Allen, Mr. William Henry    male   
..                                                 ...     ...   
886                              Montvila, Rev. Juozas    male   
887                       Graham, Miss. Margaret Edith  female   
888           Johnston, Miss. Catherine Helen "Carrie"  female   
889                              Behr, Mr. Karl Howell    male   
890                                Dooley, Mr. Patrick    male   

               Ticket Cabin Embarked  
0           A/5 21171   NaN        S  
1            PC 17599   C85        C  
2    STON/O2. 3101282   NaN        S  
3              113803  C123        S  
4              373450   NaN        S  
..                ...   ...      ...  
886            211536   NaN        S  
887            112053   B42        S  
888        W./C. 6607   NaN        S  
889            111369  C148        C  
890            370376   NaN        Q  

[891 rows x 5 columns]

In [9]:
df[df.dtypes[df.dtypes=="object"].index]

Name     Sex  \
0                              Braund, Mr. Owen Harris    male   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female   
2                               Heikkinen, Miss. Laina  female   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female   
4                             Allen, Mr. William Henry    male   
..                                                 ...     ...   
886                              Montvila, Rev. Juozas    male   
887                       Graham, Miss. Margaret Edith  female   
888           Johnston, Miss. Catherine Helen "Carrie"  female   
889                              Behr, Mr. Karl Howell    male   
890                                Dooley, Mr. Patrick    male   

               Ticket Cabin Embarked  
0           A/5 21171   NaN        S  
1            PC 17599   C85        C  
2    STON/O2. 3101282   NaN        S  
3              113803  C123        S  
4              373450   NaN        S  
..                ...   ...      ...  
886            211536   NaN        S  
887            112053   B42        S  
888        W./C. 6607   NaN        S  
889            111369  C148        C  
890            370376   NaN        Q  

[891 rows x 5 columns]

In [10]:
df.dtypes[df.dtypes=="int64"].index

Index(['PassengerId', 'Survived', 'Pclass', 'SibSp', 'Parch'], dtype='object')

In [11]:
df[df.dtypes[df.dtypes=="int64"].index]

PassengerId  Survived  Pclass  SibSp  Parch
0              1         0       3      1      0
1              2         1       1      1      0
2              3         1       3      0      0
3              4         1       1      1      0
4              5         0       3      0      0
..           ...       ...     ...    ...    ...
886          887         0       2      0      0
887          888         1       1      0      0
888          889         0       3      1      2
889          890         1       1      0      0
890          891         0       3      0      0

[891 rows x 5 columns]

In [24]:
df[df.dtypes[(df.dtypes == "float64") | (df.dtypes== "int64" )].index]

PassengerId  Survived  Pclass   Age  SibSp  Parch     Fare
0              1         0       3  22.0      1      0   7.2500
1              2         1       1  38.0      1      0  71.2833
2              3         1       3  26.0      0      0   7.9250
3              4         1       1  35.0      1      0  53.1000
4              5         0       3  35.0      0      0   8.0500
..           ...       ...     ...   ...    ...    ...      ...
886          887         0       2  27.0      0      0  13.0000
887          888         1       1  19.0      0      0  30.0000
888          889         0       3   NaN      1      2  23.4500
889          890         1       1  26.0      0      0  30.0000
890          891         0       3  32.0      0      0   7.7500

[891 rows x 7 columns]

In [59]:
df[df.dtypes[(df.dtypes== "float64" ) | (df.dtypes== "int64" )].index]

PassengerId  Survived  Pclass   Age  SibSp  Parch     Fare
0              1         0       3  22.0      1      0   7.2500
1              2         1       1  38.0      1      0  71.2833
2              3         1       3  26.0      0      0   7.9250
3              4         1       1  35.0      1      0  53.1000
4              5         0       3  35.0      0      0   8.0500
..           ...       ...     ...   ...    ...    ...      ...
886          887         0       2  27.0      0      0  13.0000
887          888         1       1  19.0      0      0  30.0000
888          889         0       3   NaN      1      2  23.4500
889          890         1       1  26.0      0      0  30.0000
890          891         0       3  32.0      0      0   7.7500

[891 rows x 7 columns]

In [26]:
df[df.dtypes[(df.dtypes== "float64" ) |   (df.dtypes=="int64")].index]

PassengerId  Survived  Pclass   Age  SibSp  Parch     Fare
0              1         0       3  22.0      1      0   7.2500
1              2         1       1  38.0      1      0  71.2833
2              3         1       3  26.0      0      0   7.9250
3              4         1       1  35.0      1      0  53.1000
4              5         0       3  35.0      0      0   8.0500
..           ...       ...     ...   ...    ...    ...      ...
886          887         0       2  27.0      0      0  13.0000
887          888         1       1  19.0      0      0  30.0000
888          889         0       3   NaN      1      2  23.4500
889          890         1       1  26.0      0      0  30.0000
890          891         0       3  32.0      0      0   7.7500

[891 rows x 7 columns]

In [29]:
df[["Age","Parch","Survived"]][0:6]

Age  Parch  Survived
0  22.0      0         0
1  38.0      0         1
2  26.0      0         1
3  35.0      0         1
4  35.0      0         0
5   NaN      0         0

In [61]:
df["Survived"][0:6]

0    0
1    1
2    1
3    1
4    0
5    0
Name: Survived, dtype: int64

In [13]:
df["Name"][0:5]

0                              Braund, Mr. Owen Harris
1    Cumings, Mrs. John Bradley (Florence Briggs Th...
2                               Heikkinen, Miss. Laina
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)
4                             Allen, Mr. William Henry
Name: Name, dtype: object

In [14]:
df["Age"]

0      22.0
1      38.0
2      26.0
3      35.0
4      35.0
       ... 
886    27.0
887    19.0
888     NaN
889    26.0
890    32.0
Name: Age, Length: 891, dtype: float64

In [33]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [63]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [35]:
df["Age"].isnull() .sum()

177

In [36]:
import numpy as np

In [39]:
np.where(df["Age"].isnull()==True)

(array([  5,  17,  19,  26,  28,  29,  31,  32,  36,  42,  45,  46,  47,
         48,  55,  64,  65,  76,  77,  82,  87,  95, 101, 107, 109, 121,
        126, 128, 140, 154, 158, 159, 166, 168, 176, 180, 181, 185, 186,
        196, 198, 201, 214, 223, 229, 235, 240, 241, 250, 256, 260, 264,
        270, 274, 277, 284, 295, 298, 300, 301, 303, 304, 306, 324, 330,
        334, 335, 347, 351, 354, 358, 359, 364, 367, 368, 375, 384, 388,
        409, 410, 411, 413, 415, 420, 425, 428, 431, 444, 451, 454, 457,
        459, 464, 466, 468, 470, 475, 481, 485, 490, 495, 497, 502, 507,
        511, 517, 522, 524, 527, 531, 533, 538, 547, 552, 557, 560, 563,
        564, 568, 573, 578, 584, 589, 593, 596, 598, 601, 602, 611, 612,
        613, 629, 633, 639, 643, 648, 650, 653, 656, 667, 669, 674, 680,
        692, 697, 709, 711, 718, 727, 732, 738, 739, 740, 760, 766, 768,
        773, 776, 778, 783, 790, 792, 793, 815, 825, 826, 828, 832, 837,
        839, 846, 849, 859, 863, 868, 878, 888], dt

In [43]:
df.iloc[np.where(df["Age"].isnull()==True)]["Ticket"]

5          330877
17         244373
19           2649
26           2631
28         330959
          ...    
859          2629
863      CA. 2343
868        345777
878        349217
888    W./C. 6607
Name: Ticket, Length: 177, dtype: object

In [45]:
df.iloc[np.where(df["Age"].isnull()==True)][["Ticket"]]

Ticket
5        330877
17       244373
19         2649
26         2631
28       330959
..          ...
859        2629
863    CA. 2343
868      345777
878      349217
888  W./C. 6607

[177 rows x 1 columns]

In [66]:
np.where(df["Age"].isnull()==True)

(array([  5,  17,  19,  26,  28,  29,  31,  32,  36,  42,  45,  46,  47,
         48,  55,  64,  65,  76,  77,  82,  87,  95, 101, 107, 109, 121,
        126, 128, 140, 154, 158, 159, 166, 168, 176, 180, 181, 185, 186,
        196, 198, 201, 214, 223, 229, 235, 240, 241, 250, 256, 260, 264,
        270, 274, 277, 284, 295, 298, 300, 301, 303, 304, 306, 324, 330,
        334, 335, 347, 351, 354, 358, 359, 364, 367, 368, 375, 384, 388,
        409, 410, 411, 413, 415, 420, 425, 428, 431, 444, 451, 454, 457,
        459, 464, 466, 468, 470, 475, 481, 485, 490, 495, 497, 502, 507,
        511, 517, 522, 524, 527, 531, 533, 538, 547, 552, 557, 560, 563,
        564, 568, 573, 578, 584, 589, 593, 596, 598, 601, 602, 611, 612,
        613, 629, 633, 639, 643, 648, 650, 653, 656, 667, 669, 674, 680,
        692, 697, 709, 711, 718, 727, 732, 738, 739, 740, 760, 766, 768,
        773, 776, 778, 783, 790, 792, 793, 815, 825, 826, 828, 832, 837,
        839, 846, 849, 859, 863, 868, 878, 888], dt

In [70]:
df.iloc[np.where(df["Age"].isnull()==True)]["Name"]

5                              Moran, Mr. James
17                 Williams, Mr. Charles Eugene
19                      Masselmani, Mrs. Fatima
26                      Emir, Mr. Farred Chehab
28                O'Dwyer, Miss. Ellen "Nellie"
                         ...                   
859                            Razi, Mr. Raihed
863           Sage, Miss. Dorothy Edith "Dolly"
868                 van Melkebeke, Mr. Philemon
878                          Laleff, Mr. Kristo
888    Johnston, Miss. Catherine Helen "Carrie"
Name: Name, Length: 177, dtype: object

In [17]:
import numpy as np
np.where(df["Age"].isnull()==True)

(array([  5,  17,  19,  26,  28,  29,  31,  32,  36,  42,  45,  46,  47,
         48,  55,  64,  65,  76,  77,  82,  87,  95, 101, 107, 109, 121,
        126, 128, 140, 154, 158, 159, 166, 168, 176, 180, 181, 185, 186,
        196, 198, 201, 214, 223, 229, 235, 240, 241, 250, 256, 260, 264,
        270, 274, 277, 284, 295, 298, 300, 301, 303, 304, 306, 324, 330,
        334, 335, 347, 351, 354, 358, 359, 364, 367, 368, 375, 384, 388,
        409, 410, 411, 413, 415, 420, 425, 428, 431, 444, 451, 454, 457,
        459, 464, 466, 468, 470, 475, 481, 485, 490, 495, 497, 502, 507,
        511, 517, 522, 524, 527, 531, 533, 538, 547, 552, 557, 560, 563,
        564, 568, 573, 578, 584, 589, 593, 596, 598, 601, 602, 611, 612,
        613, 629, 633, 639, 643, 648, 650, 653, 656, 667, 669, 674, 680,
        692, 697, 709, 711, 718, 727, 732, 738, 739, 740, 760, 766, 768,
        773, 776, 778, 783, 790, 792, 793, 815, 825, 826, 828, 832, 837,
        839, 846, 849, 859, 863, 868, 878, 888], dt

In [18]:
df[df["Age"].isnull()==True]["Name"]

5                              Moran, Mr. James
17                 Williams, Mr. Charles Eugene
19                      Masselmani, Mrs. Fatima
26                      Emir, Mr. Farred Chehab
28                O'Dwyer, Miss. Ellen "Nellie"
                         ...                   
859                            Razi, Mr. Raihed
863           Sage, Miss. Dorothy Edith "Dolly"
868                 van Melkebeke, Mr. Philemon
878                          Laleff, Mr. Kristo
888    Johnston, Miss. Catherine Helen "Carrie"
Name: Name, Length: 177, dtype: object

In [19]:
df.iloc[[1,6,7]]

PassengerId  Survived  Pclass  \
1            2         1       1   
6            7         0       1   
7            8         0       3   

                                                Name     Sex   Age  SibSp  \
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
6                            McCarthy, Mr. Timothy J    male  54.0      0   
7                     Palsson, Master. Gosta Leonard    male   2.0      3   

   Parch    Ticket     Fare Cabin Embarked  
1      0  PC 17599  71.2833   C85        C  
6      0     17463  51.8625   E46        S  
7      1    349909  21.0750   NaN        S

In [20]:
df.iloc[np.where(df["Age"].isnull()==True)]["Name"]

5                              Moran, Mr. James
17                 Williams, Mr. Charles Eugene
19                      Masselmani, Mrs. Fatima
26                      Emir, Mr. Farred Chehab
28                O'Dwyer, Miss. Ellen "Nellie"
                         ...                   
859                            Razi, Mr. Raihed
863           Sage, Miss. Dorothy Edith "Dolly"
868                 van Melkebeke, Mr. Philemon
878                          Laleff, Mr. Kristo
888    Johnston, Miss. Catherine Helen "Carrie"
Name: Name, Length: 177, dtype: object

In [79]:
max(df["Fare"])

512.3292

In [46]:
max(df["Fare"])

512.3292

In [77]:
np.where(df["Fare"]==max(df["Fare"]))

(array([258, 679, 737], dtype=int64),)

In [21]:
np.where(df["Fare"]==max(df["Fare"]))

(array([258, 679, 737], dtype=int64),)

In [22]:
df.iloc[np.where(df["Fare"]==max(df["Fare"]))]

PassengerId  Survived  Pclass                                Name  \
258          259         1       1                    Ward, Miss. Anna   
679          680         1       1  Cardeza, Mr. Thomas Drake Martinez   
737          738         1       1              Lesurer, Mr. Gustave J   

        Sex   Age  SibSp  Parch    Ticket      Fare        Cabin Embarked  
258  female  35.0      0      0  PC 17755  512.3292          NaN        C  
679    male  36.0      0      1  PC 17755  512.3292  B51 B53 B55        C  
737    male  35.0      0      0  PC 17755  512.3292         B101        C

In [23]:
np.where(df["Sex"]=="male")

(array([  0,   4,   5,   6,   7,  12,  13,  16,  17,  20,  21,  23,  26,
         27,  29,  30,  33,  34,  35,  36,  37,  42,  45,  46,  48,  50,
         51,  54,  55,  57,  59,  60,  62,  63,  64,  65,  67,  69,  70,
         72,  73,  74,  75,  76,  77,  78,  80,  81,  83,  86,  87,  89,
         90,  91,  92,  93,  94,  95,  96,  97,  99, 101, 102, 103, 104,
        105, 107, 108, 110, 112, 115, 116, 117, 118, 120, 121, 122, 124,
        125, 126, 127, 129, 130, 131, 134, 135, 137, 138, 139, 143, 144,
        145, 146, 148, 149, 150, 152, 153, 154, 155, 157, 158, 159, 160,
        162, 163, 164, 165, 168, 169, 170, 171, 173, 174, 175, 176, 178,
        179, 181, 182, 183, 185, 187, 188, 189, 191, 193, 196, 197, 200,
        201, 202, 203, 204, 206, 207, 209, 210, 212, 213, 214, 217, 219,
        220, 221, 222, 223, 224, 225, 226, 227, 228, 231, 232, 234, 236,
        238, 239, 242, 243, 244, 245, 248, 249, 250, 252, 253, 260, 261,
        262, 263, 265, 266, 267, 270, 271, 273, 277

In [24]:
df.iloc[np.where(df["Sex"]=="male")]

PassengerId  Survived  Pclass                            Name   Sex  \
0              1         0       3         Braund, Mr. Owen Harris  male   
4              5         0       3        Allen, Mr. William Henry  male   
5              6         0       3                Moran, Mr. James  male   
6              7         0       1         McCarthy, Mr. Timothy J  male   
7              8         0       3  Palsson, Master. Gosta Leonard  male   
..           ...       ...     ...                             ...   ...   
883          884         0       2   Banfield, Mr. Frederick James  male   
884          885         0       3          Sutehall, Mr. Henry Jr  male   
886          887         0       2           Montvila, Rev. Juozas  male   
889          890         1       1           Behr, Mr. Karl Howell  male   
890          891         0       3             Dooley, Mr. Patrick  male   

      Age  SibSp  Parch            Ticket     Fare Cabin Embarked  
0    22.0      1      0         A/5 21171   7.2500   NaN        S  
4    35.0      0      0            373450   8.0500   NaN        S  
5     NaN      0      0            330877   8.4583   NaN        Q  
6    54.0      0      0             17463  51.8625   E46        S  
7     2.0      3      1            349909  21.0750   NaN        S  
..    ...    ...    ...               ...      ...   ...      ...  
883  28.0      0      0  C.A./SOTON 34068  10.5000   NaN        S  
884  25.0      0      0   SOTON/OQ 392076   7.0500   NaN        S  
886  27.0      0      0            211536  13.0000   NaN        S  
889  26.0      0      0            111369  30.0000  C148        C  
890  32.0      0      0            370376   7.7500   NaN        Q  

[577 rows x 12 columns]

1.where fare is greater than 51.8625 extract all that data

2. extract only intergers columns

In [56]:
df["Sex"].mode()

0    male
dtype: object

In [49]:
df["Age"]

0      22.0
1      38.0
2      26.0
3      35.0
4      35.0
       ... 
886    27.0
887    19.0
888     NaN
889    26.0
890    32.0
Name: Age, Length: 891, dtype: float64

In [50]:
df["Age"].mean()

29.69911764705882

In [48]:
df["Age"].fillna(df["Age"].mean())

0      22.000000
1      38.000000
2      26.000000
3      35.000000
4      35.000000
         ...    
886    27.000000
887    19.000000
888    29.699118
889    26.000000
890    32.000000
Name: Age, Length: 891, dtype: float64

In [51]:
df["Age"].dropna()

0      22.0
1      38.0
2      26.0
3      35.0
4      35.0
       ... 
885    39.0
886    27.0
887    19.0
889    26.0
890    32.0
Name: Age, Length: 714, dtype: float64

In [52]:
891-714

177